db/db_impl.cc

```diff
@@ -4759,18 +4759,18 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
       if (!cfd->mem()->IsEmpty()) {
         // We only consider active mem table, hoping immutable memtable is
         // already in the process of flushing.
-        size_t cfd_size = cfd->mem()->ApproximateMemoryUsage();
-        if (largest_cfd == nullptr || cfd_size > largest_cfd_size) {
-          largest_cfd = cfd;
-          largest_cfd_size = cfd_size;
+        uint64_t seq = cfd->mem()->GetLatestSeqWhenCreate();
```

Comment:
> it is closer to how we determine whether a log file can be dropped

The current GetLatestSeqWhenCreate() is missing one important aspect of how we determine whether a log file can be dropped: https://github.com/facebook/rocksdb/blob/5.3.fb/db/db_impl.cc#L5287-L5294

That's my only concern about GetLatestSeqWhenCreate(). It's a major concern for us, as I explained with an example. The straightforward way to fix it is to update latest_seq_when_create_ in SwitchMemtable() for empty CFs, similar to how SetLogNumber() is done; except it has to be done even if creating_new_log is false, to make it work with our walless setup.

Fwiw, if latest_seq_when_create_ is updated in SwitchMemtable() for empty CFs, I don't see any difference between using GetFirstSequenceNumber() vs GetLatestSeqWhenCreate() (vs DBImpl::MinLogNumberToKeep() too, if WAL is enabled and 2pc is disabled). Can you point out any differences?

If you're more comfortable with GetLatestSeqWhenCreate() than GetFirstSequenceNumber(), I don't mind, but we need that "optimization" in SwitchMemtable().